lib/db.js

```diff
@@ -314,17 +314,35 @@ class LimitDBRedis extends EventEmitter {
       });
   }
 
-  buildStatusResponse(bucket, prefixLength) {
-    return (key, lastDrip, remaining) => {
+  buildStatusResponse(type, bucket, prefixLength) {
+    return (key, cb) => {
+      key = key.substring(prefixLength);
       const bucketKeyConfig = this.bucketKeyConfig(bucket, key);
-      return {
-        key: key.substring(prefixLength),
-        instance: key.substring(prefixLength),
-        limit: bucketKeyConfig.size,
-        reset: this.calculateReset(bucketKeyConfig, remaining),
-        remaining
-      };
-    };
+      this.redis.hmget(`${type}:${key}`, 'd', 'r', (err, hash) => {
+        if (err) {
+          return cb(err);
+        }
+        const lastDrip = parseInt(hash[0], 10);
+        const remaining = parseInt(hash[1], 10);
+        return cb(null, {
+          key,
+          instance: key,
+          limit: bucketKeyConfig.size,
+          reset: this.calculateReset(bucketKeyConfig, remaining),
+          remaining
+        });
+      });
+    }
+  }
+
+  scanStream(params) {
+    const nodes = this.redis.nodes ? this.redis.nodes('slave') : [this.redis];
+    return mergeStream(nodes.map((node) => {
+      return node.scanStream({
```

Comment:
what happens with the errors of the individual streams? I'm not sure how merge stream works, but in general if you don't listen for them they will end up as uncaughts unless mergestream does

Reply:
Merge stream bubbles them up and listens for them individually.

https://github.com/grncdr/merge-stream/blob/master/index.js#L28